MyToolsYourToolsClient/src/app/admin-panel/notifications/notifications.component.html

```diff
@@ -19,12 +19,12 @@ <h5>Powiadomienia</h5>
               </span>
             </div>
             <div clss="col-auto">
-              <button class="btn btn-success btn-sm float-right">
+              <button (click)="onRequestNotificationApproved($event)" class="btn btn-success btn-sm float-right" id="{{request.id}}">
```

Comment:
Jak tak działa to też spoko, ja generalnie zrobiłbym:
(click)="onRequestNotificationApproved(request.id)"
bez ustawianie tego id w html, ale jak tak się da i działa to spoko, nawet nie wiedziałem że tak idzie

Reply:
właśnie chciałem zrobic tak jak piszesz ale wywalało błąd, a tak poszło więc zostawiłem

Reply:
a to zwracam honor, choć troche mnie to zastanawia

Reply:
Już wiem, bo metoda onRequestNotificationApproved(request.id) powinna przyjmować jako argument tylko requestId: string, coś takiego:
  onRequestNotificationApproved(requestId: string){

    this.rentRequests = this.rentRequests.filter(n => n.id !== requestId);
    this.notificationService.deleteNotification(requestId);

  }
Wtedy mi zadziałało ;)
Jak powyzsze też działa to już zostawmy, ale na przyszłość tak jak co, bo to bardziej intuicyjne jest, tych event-emiterów raczej się do komunikacji między komponentami używa